NSubstitute.DbConnection/NSubstitute.DbConnection/MockQuery.cs

```diff
@@ -194,5 +244,22 @@ private static QueryInfo GetQueryInfo(IDbCommand mockCommand)
             };
             return queryInfo;
         }
+
+        private static void SetupOutputParams(IDbCommand mockCommand, Dictionary<string, QueryParameter> parameters)
+        {
+            for (var i = 0; i < mockCommand.Parameters.Count; i++)
+            {
+                var cmdParam = mockCommand.Parameters[i];
+                if (cmdParam is DbParameter dbParameter && dbParameter.Direction == ParameterDirection.Output)
+                {
+                    if (!parameters.TryGetValue(dbParameter.ParameterName, out var value))
+                    {
+                        throw new NotSupportedException($"Unmatched output parameter: '{mockCommand.CommandText}'");
+                    }
+
+                    dbParameter.Value = value.Value;
```

Comment:
These `Setup*()` methods are generally busy setting up mock return values for methods on the reader and so on. Am I right in thinking here that we can just use the concrete DbParameter object "as is", without needing to mock it, which is why it's okay to just set a property value rather than set up the return value of a mock object?

Reply:
So we are using the concrete `DbParameter` type ([DbConnnectionMockExtensions.cs#97](https://github.com/jmg48/NSubstitute.DbConnection/blob/852189ef0c5f42481ecfa93617b79f08ad7b1f64/NSubstitute.DbConnection/NSubstitute.DbConnection/DbConnnectionMockExtensions.cs#L97)) but mocking the `DbParameterCollection` object on the command. I assume because `DbParameterCollection` is abstract?